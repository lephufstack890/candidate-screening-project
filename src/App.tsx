import React, { useState, useEffect } from 'react';
import './App.css';
import ListItem from './components/ListItem';
import Add from './components/Add';
import * as ListItemServices from './services/ListItemService';
import { IState } from './App.types';

function App() {

  const [items, setItems] = useState<IState['items']>([{ name: '', quantity: 1, checked: false }]);

  useEffect(() => {
    reloadData();
  }, []);
  
  const reloadData = async () => {
      const result = await ListItemServices.list();
      setItems(result);
  };

  return (
    <div className="App">
      <div className="App__wrapper">
        <Add reloadData={reloadData} />
        <ListItem items={items} setItems={setItems} reloadData={reloadData} />
      </div>
    </div>
  );
}

export default App;
