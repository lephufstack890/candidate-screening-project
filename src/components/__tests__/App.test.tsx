// import React from 'react';
// import { render, screen } from '@testing-library/react';
// import App from '../../App';

// test('renders learn react link', () => {
//   // expect(true).toBe(true)
//   render(<App />);
//   // const linkElement = screen.getByText(/learn react/i);
//   // expect(linkElement).toBeInTheDocument();
// });

import renderer from 'react-test-renderer';
import App from '../../App';
// describe("snapshot testing", () => {
//   expect(true).toBe(true)
// })

describe('my test', () => {
  it('some text', () => {
    const renderedComponent = renderer.create(<App />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
});
