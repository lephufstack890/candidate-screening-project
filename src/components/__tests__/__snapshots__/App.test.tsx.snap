// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my test some text 1`] = `
<div
  className="App"
>
  <div
    className="App__wrapper"
  >
    <form
      action="#"
      autoComplete="off"
      className="add__form"
      noValidate={true}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Add an item...
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Add an item...
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="add__button"
        type="submit"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium add__icon css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
    </form>
    <div>
      <div
        className="ListItem"
      >
        <div
          className="ListItem__name"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <span
            className="ListItem__nocompleted"
          />
        </div>
        <div
          className="ListItem__quantity"
        >
          <button
            className="ListItem__button-reduce"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
          <span>
            1
          </span>
          <button
            className="ListItem__button-increase"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="App__total"
      >
        Total: 
        1
      </div>
    </div>
  </div>
</div>
`;
